### PR TITLE
CMR-6288 - Spatial plugin to work with elasticsearch 7.5.2.

### PR DESCRIPTION
**Pull request by CayvonH**
* Update spatial plugin to use new interfaces provided by es 7.5.2.
* Use embedded docker to run elasticsearch instead of embedded elasticsearch. Necessary to avoid jar hell checks and install plugin gracefully. 
* Add install spatial plugin to cmr script.

### CODE REVIEW

**Comment by ygliuvt** [comment]
We need to make sure this branch run tests successfully in CI.

File: dev-system/project.clj [hunk 1]
```diff
@@ -135,10 +135,16 @@
...
             ["test2junit"]
-            ;; Installs the Elasticsearch Marvel plugin locally.
-            ;; Visit http://localhost:9210/_plugin/marvel/sense/index.html
-            "install-marvel"
-            ["shell" "cmr" "install" "local" "marvel"]
+            ;; Install spatial plugin locally
+            "install-spatial-plugin"
+            ["shell" "cmr" "install" "local" "spatial_plugin"]
+            ;; Install with local elasticsearch
+            "install"
```
Comment (ygliuvt) [inline dev-system/project.clj:142]: Shouldn't this be `"install!"` and with line 146 and 147 removed?

File: elastic-utils-lib/src/cmr/elastic_utils/embedded_elastic_server.clj [hunk 2]
```diff
@@ -1,75 +1,124 @@
...
+           (.withEnv "indices.breaker.total.use_real_memory" "false")
+           (.withEnv "node.name" "embedded-elastic")
+           (.withNetwork network)
+           (.withNetworkAliases (into-array String ["elasticsearch"]))
+           (.withFixedExposedPort (int http-port) 9200)
+           (.withStartupTimeout (Duration/ofSeconds 120))
+           (.waitingFor
+            (.forStatusCode (Wait/forHttp "/_cat/health?v&pretty") 200)))
+     {:elasticsearch container
+      :kibana kibana})))
```
Comment (ygliuvt) [inline elastic-utils-lib/src/cmr/elastic_utils/embedded_elastic_server.clj:75]: It is nice that you are adding Kibana to the elasticsearch node as well.

File: es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj [hunk 3]
```diff
@@ -1,30 +1,87 @@
 (ns cmr.elasticsearch.plugins.spatial.script.core
+  (:require
+   [clojure.string :as s]
```
Comment (ygliuvt) [inline es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj:3]: remove unused namespace reference.

File: es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj [hunk 4]
```diff
@@ -1,30 +1,87 @@
 (ns cmr.elasticsearch.plugins.spatial.script.core
```
Comment (ygliuvt) [inline es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj:1]: There are some incredible code in this file. Good job!

File: dev-system/project.clj (hunk 1 above)
Comment (CayvonH) [reply dev-system/project.clj:142]: Hmm i was thinking we wanted the plugin to install with `lein install`. Think i should change so it only gets installed with `lein install!`?

File: es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj (hunk 3 above)
Comment (CayvonH) [reply es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj:3]: done

File: dev-system/project.clj (hunk 1 above)
Comment (ygliuvt) [reply dev-system/project.clj:142]: nm. I think you are right.

File: es-spatial-plugin/project.clj [hunk 5]
```diff
@@ -13,59 +13,69 @@
...
+            [test2junit "1.3.3"]]
   :profiles {:security {:plugins [[com.livingsocial/lein-dependency-check "1.1.1"]]
                         :dependency-check {:output-format [:all]
                                            :suppression-file "resources/security/suppression.xml"}}
+             :provided {:dependencies [[nasa-cmr/cmr-common-lib "0.1.1-SNAPSHOT"]
+                                       [nasa-cmr/cmr-spatial-lib "0.1.0-SNAPSHOT"]
+                                       [org.elasticsearch/elasticsearch "7.5.2"]]}
+             :es-deps {:dependencies [[nasa-cmr/cmr-spatial-lib "0.1.0-SNAPSHOT"
+                                       ;; These exclusions will be provided by elasticsearch.
+                                       :exclusions [[com.dadrox/quiet-slf4j]
```
Comment (ygliuvt) [inline es-spatial-plugin/project.clj:40]: Man, all these exclusions!

Comment (ygliuvt) [reply es-spatial-plugin/project.clj:40]: Hopefully, we don't have to keep updating these when ES is upgraded.

**Comment by ygliuvt** [comment]
What's happening with the build? It is not running in CI.

File: es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/factory/lfactory.clj [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  [params]
+  (when-not (every? params parameters)
+    (throw (IllegalArgumentException.
+             (str "Missing one or more of required parameters: "
+                  (clojure.string/join parameters ", "))))))
+
+(defn- convert-params
+  "Convert the comma separated string into a vector of integers."
+  [[k v]]
+  [k (map #(Integer. ^String %) (str/split v #","))])
```
Comment (daniel-zamora) [inline es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/factory/lfactory.clj:50]: We will get some exceptions here, do we validate the params well enough to not care?

File: es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj [hunk 7]
```diff
@@ -1,30 +1,86 @@
...
+(defn- get-from-fields
+  [^LeafFieldsLookup lookup key]
+  (when (and lookup key (.containsKey lookup key))
+    (when-let [^FieldLookup field-lookup (.get lookup key)]
+      (seq (.getValues field-lookup)))))
+
+(defn doc-intersects?
+  "Returns true if the doc contains a ring that intersects the ring passed in."
+  [^LeafFieldsLookup lookup intersects-fn]
+  ; Must explicitly return true or false or elastic search will complain
```
Comment (daniel-zamora) [inline es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj:40]: Two ";" for single line comments.

Comment (CayvonH) [reply es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj]: done

**Pull request merged**